navpy/core/navpy.py

```diff
@@ -903,33 +903,37 @@ def ecef2lla(ecef, latlon_unit='deg'):
     alt : {(N,)} array like altitude in meters
     """
     ecef,N = _input_check_Nx3(ecef)
-    if(N>1):
-        x = ecef[:,0]; y = ecef[:,1]; z = ecef[:,2]
-    else:
-        x = ecef[0]; y = ecef[1]; z = ecef[2]
-
+    ecef = ecef.reshape(N,3)
+    x = ecef[:,0]; y = ecef[:,1]; z = ecef[:,2]
+    
     lon = np.arctan2(y,x)
 
     # Iteration to get Latitude and Altitude
     p = np.sqrt(x**2 + y**2)
     lat = np.arctan2(z,p*(1-wgs84._ecc_sqrd))
 
     err = np.ones(N)
-
+    h = np.zeros(N)
     while(np.max(np.abs(err))>1e-10):
         Rew,Rns = earthrad(lat,lat_unit='rad')
-        h = p/np.cos(lat) - Rew
         
-        err = np.arctan2(z*(1+wgs84._ecc_sqrd*Rew*np.sin(lat)/z),p) - lat
+        idx = (np.pi/2*np.ones(N) - np.abs(lat)) > 1e-3
+        # For lat < 90 degrees
+        h[idx] = np.divide(p[idx],np.cos(lat[idx])) - Rew[idx]
+        # For lat == 90 degrees
+        h[~idx] = np.divide(z[~idx],np.sin(lat[~idx])) - (1-wgs84._ecc_sqrd)*Rew[~idx]
```

Comment:
To be complete can the comment be updated to:
1. Specify the first case is for lat << 90 deg
2. Specify the second case is for lat ~90 deg
3. Add a comment that the first case use the original reference and the second case uses the inverse of the altitude expression in lla2ecef

Reply:
After this I'll go ahead and merge.  Thanks for adding these extra test cases for the poles as well @adhika .

Reply:
This is minor enough and not worth holding back the PR.  I've moved it to an issue (#21)  and will merge this PR.